### PR TITLE
bugfix_sup-46_warehouse_page

### PR DESCRIPTION
**Pull request by alexeyoverchenko**

### CODE REVIEW

**Review by IrynaZazyba — APPROVED** [review]

File: frontend/src/pages/warehouse/ModalAddWarehouse.js [hunk 1]
```diff
@@ -127,63 +127,67 @@ function ModalAddWarehouse(props) {
 
     const addWarehouseHandler = (e) => {
```
Comment (eugenmolchanov) [inline frontend/src/pages/warehouse/ModalAddWarehouse.js:128]: too big and nested function. Please, divide it into smaller ones.

File: frontend/src/pages/warehouse/ModalEditWarehouse.js [hunk 2]
```diff
@@ -137,49 +137,55 @@ function ModalEditWarehouse(props) {
...
         e.preventDefault();
-        let location = `${warehouseDto.addressDto.state.state} 
+
+        let warehouseUpdateDto = {};
+        let dispatchersId = dispatchers.map(dispatcher => dispatcher.id);
+        warehouseUpdateDto = {
+            ...warehouseDto,
+            dispatchersId: dispatchersId,
+            customerId: props.currentCustomerId,
+            irrelevantDispatchersId: dispatcherDeleteList
```
Comment (eugenmolchanov) [inline frontend/src/pages/warehouse/ModalEditWarehouse.js:147]: if this list is derived from some other list in the state or any other place, we could have a function defined in the component, which give use this list whenever we want. We don't need to keep 2 lists in the state, if one of them can be derived from another one.

**Review by eugenmolchanov — CHANGES_REQUESTED** [review]
Please, see the comments.

File: frontend/src/pages/warehouse/ModalAddWarehouse.js (hunk 1 above)
Comment (alexeyoverchenko) [reply frontend/src/pages/warehouse/ModalAddWarehouse.js:128]: got it, thanks!

**Review by eugenmolchanov — APPROVED** [review]

**Pull request merged**